azure/storage/blob/baseblobservice.py

```diff
@@ -244,11 +244,13 @@ def make_blob_url(self, container_name, blob_name,
             blob_name,
         )
 
-        if snapshot:
-            url += '?snapshot=' + snapshot
-
-        if sas_token:
-            url += '?' + sas_token
+        if snapshot and sas_token:
+            url += '?snapshot=' + snapshot + '&' + sas_token
+        else:
+            if snapshot:
+                url += '?snapshot=' + snapshot
+            elif sas_token:
+                url += '?' + sas_token
```

Comment:
Looks good to me, but following form can save memory allocation as well reduce indention.

``` python
if snapshot and sas_token:
    url = '{}?snapshot={}&{}'.format(url, snapshot, sas)
elif snapshot:
    url = '{}?snapshot={}'.format(url, snapshot)
elif sas_token:
    url = '{}?{}'.format(url, sas)